src/utils.cc

```diff
@@ -740,14 +757,31 @@ bool utils::anyParameterAtBoundaries( const RooArgSet &params, int verbosity ){
 
     RooLinkedListIter iter = params.iterator(); int i = 0;
     for (RooRealVar *a = (RooRealVar *) iter.Next(); a != 0; a = (RooRealVar *) iter.Next(), ++i) {
-
-        bool isBad = isParameterAtBoundary(*a);
+      
+        bool atLowerBoundary = isParameterAtLowerBoundary(*a);
+        bool atUpperBoundary = isParameterAtLowerBoundary(*a);
+        bool isBad = atLowerBoundary || atUpperBoundary;
 
         if(isBad){
             std::string varName((*a).GetName());
 
             if( verbosity >= 9 || (timesFoundAtBoundary[varName] < 3 && verbosity > -1) ){
-                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at boundary. \n", (*a).GetName());
+              
+                fprintf(CloseCoutSentry::trueStdOutGlobal(),"  [WARNING] Found [%s] at ", (*a).GetName());
+
+                if (atLowerBoundary && atUpperBoundary)
+                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"both boundaries");
+                else if (atLowerBoundary)
+                  fprintf(CloseCoutSentry::trueStdOutGlobal(),"lower boundary");
+                else 
+                  // must be at upper boundary
```

Comment:
It would be better to not trust, but rather test everything and if not fitting any case, give some meaningful message.